File: 백준/Silver/11047. 동전 0/동전 0.js
const fs = require('fs');
const input = fs.readFileSync(0, 'utf-8').toString().trim().split('\n');
const [N, K] = input[0].split(' ').map(Number);
const coins = input.slice(1).map(Number);
let money = K;
let answer = 0;

while (money > 0) {
    const coin = coins.pop();
    answer += parseInt(money / coin);
    money %= coin;
}

console.log(answer);

File: 백준/Silver/11047. 동전 0/동전 0.py
import sys

N, K = map(int,input().split())
C = [int(x) for x in sys.stdin.readlines()]
count = 0
coin_idx = N-1

while coin_idx >= 0:
    coin = C[coin_idx]
    count += K // coin
    K = K % coin
    coin_idx -= 1
    
print(count)